src/main/java/org/prlprg/rds/RDSLogger.java

```diff
@@ -0,0 +1,116 @@
+package org.prlprg.rds;
+
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.logging.*;
```

Comment:
Maybe you can use something like `Log4j 2`, or `SLF4J`and `Logback` ; it would have better performance and it is an easy swap.Log4j 2 can be easily configured dynamically with a xml file for instance.

 https://www.baeldung.com/java-logging-intro